### ISSUE TITLE
Using JavaScript delete operator has warning "expected an assignment or function call"

### ISSUE THREAD

**Issue by nmalaguti** [palantir/tslint#132]
Acceptable javascript:

``` javascript
delete obj.key;
```

Gives warning "expected an assignment or function call."

Setting `no-unused-expression = false` hides the error.
